### PR TITLE
Add Square customer creation on user registration

### PR DESCRIPTION
**Pull request by sahamanish99**

### CODE REVIEW

File: src/routers/RegisterRouter.ts [hunk 1]
```diff
@@ -17,20 +23,32 @@ class RegisterRouter extends ApplicationRouter<Object> {
 
   async content(req: Request): Promise<Object> {
     const { email, password, firstName, lastName, phoneNumber } = req.body;
+    console.log(req.body);
```
Comment (kevinchan159) [inline src/routers/RegisterRouter.ts]: remove if unncessary

File: src/routers/RegisterRouter.ts [hunk 2]
```diff
@@ -17,20 +23,32 @@ class RegisterRouter extends ApplicationRouter<Object> {
...
+          const customer = await customersApi.createCustomer(customerBody);
+          const user = await UsersRepo.createUser({
+            email,
+            password,
+            firstName,
+            lastName,
+            phoneNumber: formattedPhoneNumber,
+            customerId: customer.customer.id,
+          });
+          return (await UsersRepo.getUserById(user.id));
```
Comment (kevinchan159) [inline src/routers/RegisterRouter.ts]: is `()` necessary?

File: src/routers/RegisterRouter.ts (hunk 1 above)
Comment (sahamanish99) [reply src/routers/RegisterRouter.ts]: oh woops, resolving that asap

File: src/routers/RegisterRouter.ts (hunk 2 above)
Comment (sahamanish99) [reply src/routers/RegisterRouter.ts]: hmm i'm not sure will check

File: src/routers/RegisterRouter.ts [hunk 3]
```diff
@@ -2,10 +2,16 @@ import { randomBytes } from 'crypto';
 import { validate } from 'email-validator';
 import { Request, Response } from 'express';
 import { parsePhoneNumberFromString } from 'libphonenumber-js';
+import squareConnect from 'square-connect';
```
Comment (alanna-zhou) [inline src/routers/RegisterRouter.ts]: are these imports supposed to be alphabetized according to the words following `import` or `from`?

File: src/routers/RegisterRouter.ts [hunk 4]
```diff
@@ -21,16 +27,27 @@ class RegisterRouter extends ApplicationRouter<Object> {
...
+            phone_number: formattedPhoneNumber,
+          };
+          const customer = await customersApi.createCustomer(customerBody);
+          const user = await UsersRepo.createUser({
+            email,
+            password,
+            firstName,
+            lastName,
+            phoneNumber: formattedPhoneNumber,
+            customerId: customer.customer.id,
```
Comment (alanna-zhou) [inline src/routers/RegisterRouter.ts]: why do you have to do `customer.customer.id` and not just `customer.id`

Comment (sahamanish99) [reply src/routers/RegisterRouter.ts]: `customer.customer` gives me the contents of the actual `Customer` object from square. The key value pairs needed are nested inside a `"customer"` key with square's `createCustomer` method

File: src/routers/RegisterRouter.ts (hunk 3 above)
Comment (sahamanish99) [reply src/routers/RegisterRouter.ts]: not sure
@meganle @young-k thoughts?

Comment (kq-li) [reply src/routers/RegisterRouter.ts]: `from` according to tslint

File: src/routers/LoginRouter.ts [hunk 5]
```diff
@@ -14,7 +14,7 @@ class LoginRouter extends ApplicationRouter<Object> {
 
   async content(req: Request): Promise<Object> {
     const { email, password } = req.body;
-    return UsersRepo.authUser(email, password);
+    return await UsersRepo.authUser(email, password);
```
Comment (kq-li) [inline src/routers/LoginRouter.ts]: unnecessary await

File: src/routers/RegisterRouter.ts [hunk 6]
```diff
@@ -2,10 +2,16 @@ import { randomBytes } from 'crypto';
 import { validate } from 'email-validator';
 import { Request, Response } from 'express';
 import { parsePhoneNumberFromString } from 'libphonenumber-js';
+import squareConnect from 'square-connect';
 
 import ApplicationRouter from '../appdev/ApplicationRouter';
 import UsersRepo from '../repos/UsersRepo';
 
+const customersApi = new squareConnect.CustomersApi();
```
Comment (kq-li) [inline src/routers/RegisterRouter.ts]: `customersAPI`

File: src/routers/RegisterRouter.ts [hunk 7]
```diff
@@ -21,16 +27,27 @@ class RegisterRouter extends ApplicationRouter<Object> {
...
+          const customer = await customersApi.createCustomer(customerBody);
+          const user = await UsersRepo.createUser({
+            email,
+            password,
+            firstName,
+            lastName,
+            phoneNumber: formattedPhoneNumber,
+            customerId: customer.customer.id,
+          });
+          return await UsersRepo.getUserById(user.id);
```
Comment (kq-li) [inline src/routers/RegisterRouter.ts]: unnecessary await

File: src/routers/RegisterRouter.ts [hunk 8]
```diff
@@ -21,16 +27,27 @@ class RegisterRouter extends ApplicationRouter<Object> {
...
-        return UsersRepo.getUserById(user.id);
+        try {
+          const customerBody = {
+            idempotency_key: randomBytes(12).toString('hex'),
+            email_address: email,
+            family_name: lastName,
+            given_name: firstName,
+            phone_number: formattedPhoneNumber,
+          };
+          const customer = await customersApi.createCustomer(customerBody);
```
Comment (kq-li) [inline src/routers/RegisterRouter.ts]: think we can inline the customer body here

File: src/routers/RegisterRouter.ts [hunk 9]
```diff
@@ -21,16 +27,27 @@ class RegisterRouter extends ApplicationRouter<Object> {
...
-          firstName,
-          lastName,
-          phoneNumber: formattedPhoneNumber,
-          // TODO: Get customerId from Square API
-          customerId: randomBytes(32).toString('hex'),
-        });
-        return UsersRepo.getUserById(user.id);
+        try {
+          const customerBody = {
+            idempotency_key: randomBytes(12).toString('hex'),
```
Comment (kq-li) [inline src/routers/RegisterRouter.ts]: does this have to be 12 bytes? you can use `utils.generateToken` for a random hex string

File: src/routers/LoginRouter.ts (hunk 5 above)
Comment (sahamanish99) [reply src/routers/LoginRouter.ts]: do we need `await` for any UsersRepo method calls then? (like in RegisterRouter)

Comment (kq-li) [reply src/routers/LoginRouter.ts]: for things that aren't `return` yes, i changed a few in my PR

File: src/routers/RegisterRouter.ts (hunk 8 above)
Comment (sahamanish99) [reply src/routers/RegisterRouter.ts]: sure

File: src/routers/RegisterRouter.ts (hunk 9 above)
Comment (sahamanish99) [reply src/routers/RegisterRouter.ts]: I don't think it matters, but I will leave it for now since using utils in my branch will create a build error atm

**Pull request merged**